core/analytics/src/prod/java/com/google/samples/apps/nowinandroid/core/analytics/FirebaseAnalyticsHelper.kt

```diff
@@ -16,19 +16,17 @@
 
 package com.google.samples.apps.nowinandroid.core.analytics
 
+import com.google.firebase.analytics.FirebaseAnalytics
 import com.google.firebase.analytics.ktx.analytics
 import com.google.firebase.analytics.ktx.logEvent
-import com.google.firebase.ktx.Firebase
 import javax.inject.Inject
-import javax.inject.Singleton
 
 /**
  * Implementation of `AnalyticsHelper` which logs events to a Firebase backend.
  */
-@Singleton
 class FirebaseAnalyticsHelper @Inject constructor() : AnalyticsHelper {
 
-    private val firebaseAnalytics = Firebase.analytics
+    @Inject lateinit var firebaseAnalytics: FirebaseAnalytics
```

Comment:
If you `@Inject` a parameter, you're declaring an implicit dependency, that's an anti-pattern. 

Please, pass this dependency in the constructor so that it becomes explicit

```
class FirebaseAnalyticsHelper @Inject constructor(
    private val firebaseAnalytics: FirebaseAnalytics
) : AnalyticsHelper { }
```